### PR TITLE
Issue 29387 remove savepublish ia actionlet image

### LINKED ISSUE #29387: Remove save and publish functionality from the AI Actionlets
### Parent Issue

https://github.com/dotCMS/core/issues/29380

### User Story

As an user I want to be able to decide which is the next sub action after the IA generation

### Acceptance Criteria

Currently the IA WF actionlets does save + publish, the idea would be to leave the ability to the user to decide if wants or not save + publish or something else.

### Proposed Objective

Core Features

### Proposed Priority

Priority 3 - Average

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by jdotcms**
Removing the save + publish from the IA Actionlet Image

This PR fixes: #29387

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/ai/api/ChatAPIProvider.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package com.dotcms.ai.api;
+
+public interface ChatAPIProvider {
+
+    ChatAPI getChatAPI(Object... initArguments);
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/ai/api/ChatAPIProvider.java:5]: javadoc

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/ai/api/DotAIAPI.java [hunk 2]
```diff
@@ -9,4 +9,8 @@ public interface DotAIAPI {
     CompletionsAPI getCompletionsAPI(Object... initArguments);
 
     EmbeddingsAPI getEmbeddingsAPI(Object... initArguments);
+
+    ChatAPI getChatAPI(Object... initArguments);
```
Comment (freddyDOTCMS) [inline dotCMS/src/main/java/com/dotcms/ai/api/DotAIAPI.java:13]: javadoc

File: dotCMS/src/main/java/com/dotcms/ai/api/DotAIAPI.java [hunk 3]
```diff
@@ -9,4 +9,8 @@ public interface DotAIAPI {
...
+
+    ImageAPI getImageAPI(Object... initArguments);
```
Comment (freddyDOTCMS) [inline dotCMS/src/main/java/com/dotcms/ai/api/DotAIAPI.java:15]: javadoc

File: dotCMS/src/main/java/com/dotcms/ai/api/DotAIAPIFacadeImpl.java [hunk 4]
```diff
@@ -35,6 +45,38 @@ private static <T> T unwrap(final Class<T> clazz, final Object... initArguments)
...
+        public ChatAPI getChatAPI(final Object... initArguments) {
+            if (Objects.nonNull(initArguments) && initArguments.length > 0 && initArguments[0] instanceof AppConfig) {
+                return new OpenAIChatAPIImpl((AppConfig) initArguments[0]);
+            }
+
+            throw new IllegalArgumentException("To create a ChatAPI you need to provide an AppConfig");
+        }
+    }
+
+    public static class DefaultImageAPIProvider implements ImageAPIProvider {
```
Comment (freddyDOTCMS) [inline dotCMS/src/main/java/com/dotcms/ai/api/DotAIAPIFacadeImpl.java:60]: javadoc

File: dotCMS/src/main/java/com/dotcms/ai/api/DotAIAPIFacadeImpl.java [hunk 5]
```diff
@@ -35,6 +45,38 @@ private static <T> T unwrap(final Class<T> clazz, final Object... initArguments)
                 && clazz.isInstance(initArguments[0]) ? clazz.cast(initArguments[0]) : null;
     }
 
+    public static class DefaultChatAPIProvider implements ChatAPIProvider {
```
Comment (freddyDOTCMS) [inline dotCMS/src/main/java/com/dotcms/ai/api/DotAIAPIFacadeImpl.java:48]: javadoc

**Review by freddyDOTCMS — DISMISSED** [review]

**Review by victoralfaro-dotcms — APPROVED** [review]

**Pull request merged**